--- kernel.spec.orig	2021-11-22 10:19:21.000000000 -0500
+++ kernel.spec	2021-11-22 16:09:45.667676579 -0500
@@ -136,7 +136,7 @@
 %define patchlevel 15
 
 # allow pkg_release to have configurable %%{?dist} tag
-%define specrelease 200%{?buildid}%{?dist}
+%define specrelease 201.pixelbook%{?buildid}%{?dist}
 
 %define pkg_release %{specrelease}
 
@@ -841,7 +841,7 @@
 Source4002: gating.yaml
 
 ## Patches needed for building this package
-
+Patch0: reversed-ASoC-Intel-Skylake-Select-proper-format-for-NHLT-blob.patch
 %if !%{nopatches}
 
 Patch1: patch-%{patchversion}-redhat.patch
@@ -1379,7 +1380,7 @@
 
 cd linux-%{KVERREL}
 cp -a %{SOURCE1} .
-
+ApplyOptionalPatch reversed-ASoC-Intel-Skylake-Select-proper-format-for-NHLT-blob.patch
 %if !%{nopatches}
 
 ApplyOptionalPatch patch-%{patchversion}-redhat.patch

